src/plugins/intel_gpu/src/kernel_selector/core/cl_kernels/roi_align_ref.cl

```diff
@@ -5,99 +5,103 @@
 #include "include/batch_headers/data_types.cl"
 
-#define MAX(a,b) ((a) > (b) ? (a) : (b))
-#define NUM_ROIS OUTPUT_BATCH_NUM
-#define NUM_CHANNELS INPUT0_FEATURE_NUM
-#define POOLED_WIDTH OUTPUT_SIZE_X
+#define MAX(a, b)     ((a) > (b) ? (a) : (b))
+#define NUM_ROIS      OUTPUT_BATCH_NUM
+#define NUM_CHANNELS  INPUT0_FEATURE_NUM
+#define POOLED_WIDTH  OUTPUT_SIZE_X
 #define POOLED_HEIGHT OUTPUT_SIZE_Y
 
 KERNEL(roi_align_ref)
-(
-    const __global INPUT0_TYPE * src_data,
-    __global OUTPUT_TYPE * dst_data,
-    const __global INPUT1_TYPE * src_rois,
-    const __global INPUT2_TYPE * src_batches
-)
-{
+(const __global INPUT0_TYPE* src_data,
+ __global OUTPUT_TYPE* dst_data,
+ const __global INPUT1_TYPE* src_rois,
+ const __global INPUT2_TYPE* src_batches) {
     const size_t i = get_global_id(0);
 
     const uint x = i % POOLED_WIDTH;
     const uint y = i / POOLED_WIDTH % POOLED_HEIGHT;
     const uint c = i / POOLED_WIDTH / POOLED_HEIGHT % NUM_CHANNELS;
     const uint r = i / POOLED_WIDTH / POOLED_HEIGHT / NUM_CHANNELS % NUM_ROIS;
-    
-    const __global INPUT1_TYPE* roi_ptr = &src_rois[INPUT1_BATCH_PITCH * r];
-    
+
+    const __global INPUT1_TYPE* roi_ptr = &src_rois[INPUT1_GET_INDEX(r, 0, 0, 0)];
+
     // Get ROI`s corners
-    const INPUT1_TYPE x1 = *roi_ptr * (INPUT1_TYPE) SPATIAL_SCALE;
-    const INPUT1_TYPE y1 = roi_ptr[1] * (INPUT1_TYPE) SPATIAL_SCALE;
-    const INPUT1_TYPE x2 = roi_ptr[2] * (INPUT1_TYPE) SPATIAL_SCALE;
-    const INPUT1_TYPE y2 = roi_ptr[3] * (INPUT1_TYPE) SPATIAL_SCALE;
-    
-    const INPUT1_TYPE roi_width = MAX(x2 - x1, (INPUT1_TYPE) 1.0);
-    const INPUT1_TYPE roi_height = MAX(y2 - y1, (INPUT1_TYPE) 1.0);
+    const INPUT1_TYPE x1 =
+        (roi_ptr[0] + (INPUT1_TYPE)OFFSET_SRC) * (INPUT1_TYPE)SPATIAL_SCALE + (INPUT1_TYPE)OFFSET_DST;
+    const INPUT1_TYPE y1 =
+        (roi_ptr[1] + (INPUT1_TYPE)OFFSET_SRC) * (INPUT1_TYPE)SPATIAL_SCALE + (INPUT1_TYPE)OFFSET_DST;
+    const INPUT1_TYPE x2 =
+        (roi_ptr[2] + (INPUT1_TYPE)OFFSET_SRC) * (INPUT1_TYPE)SPATIAL_SCALE + (INPUT1_TYPE)OFFSET_DST;
+    const INPUT1_TYPE y2 =
+        (roi_ptr[3] + (INPUT1_TYPE)OFFSET_SRC) * (INPUT1_TYPE)SPATIAL_SCALE + (INPUT1_TYPE)OFFSET_DST;
+
+
+    const INPUT1_TYPE roi_width = MAX(x2 - x1, (INPUT1_TYPE)MIN_SIZE);
+    const INPUT1_TYPE roi_height = MAX(y2 - y1, (INPUT1_TYPE)MIN_SIZE);
 
     const INPUT1_TYPE bin_width = roi_width / POOLED_WIDTH;
     const INPUT1_TYPE bin_height = roi_height / POOLED_HEIGHT;
 
-    const int sampling_ratio_x = SAMPLING_RATIO == 0 ? (int) ceil(bin_width) : SAMPLING_RATIO;
-    const int sampling_ratio_y = SAMPLING_RATIO == 0 ? (int) ceil(bin_height) : SAMPLING_RATIO;
+    const int sampling_ratio_x = SAMPLING_RATIO == 0 ? (int)ceil(bin_width) : SAMPLING_RATIO;
+    const int sampling_ratio_y = SAMPLING_RATIO == 0 ? (int)ceil(bin_height) : SAMPLING_RATIO;
+
+    const INPUT1_TYPE sample_distance_x = bin_width / (INPUT1_TYPE)sampling_ratio_x;
```

Comment:
why do we need int variables ratio_x if finally convert them to float and not use them as ints anywhere?

Reply:
actually we use it as ints below in the code